src/test/java/com/parkit/parkingsystem/FareCalculatorServiceTest.java

```diff
@@ -124,4 +124,32 @@ public void calculateFareCarWithMoreThanADayParkingTime(){
         assertEquals( (24 * Fare.CAR_RATE_PER_HOUR) , ticket.getPrice());
     }
 
+    @Test
+    public void calculateFareCarWithLessThan30MinutesParkingTime() {
+        Date inTime = new Date();
+        inTime.setTime(System.currentTimeMillis() - (25* 60 * 1000)); // 25min parking time should be free
+        Date outTime = new Date();
+        ParkingSpot parkingSpot = new ParkingSpot(1, ParkingType.CAR, false);
+
+        ticket.setInTime(inTime);
+        ticket.setOutTime(outTime);
+        ticket.setParkingSpot(parkingSpot);
+        fareCalculatorService.calculateFare(ticket);
+        assertEquals(0, ticket.getPrice());
+    }
+
+    @Test
+    public void calculateFareBikeWithLessThan30MinutesParkingTime() {
+        Date inTime = new Date();
+        inTime.setTime(System.currentTimeMillis() - (25 * 60 * 1000)); // 25min parking time should be free
```

Comment:
```suggestion
        inTime.setTime(System.currentTimeMillis() - (29.9999 * 60 * 1000)); // 29min parking time should be free
```
Comme ça tu fais le test vraiment borderline.